pkg/api/pod/util.go

```diff
@@ -813,3 +818,50 @@ func setsWindowsHostProcess(podSpec *api.PodSpec) bool {
 
 	return inUse
 }
+
+func ValidatePodTemplateSpecAllowInValidLabelValueInSelector(spec *api.PodSpec) bool {
+	if spec.Affinity != nil {
+		return ValidateAffinityAllowInValidLabelValueInSelector(spec.Affinity)
+	}
+	return false
+}
+
+func ValidateAffinityAllowInValidLabelValueInSelector(affinity *core.Affinity) bool {
+	if affinity.PodAffinity != nil {
+		if affinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
no need to check for nil on slices before iterating, range is a no-op on a nil slice (applies in several places)

Reply:
The `Affinity.PodAffinity` and `Affinity.PodAntiAffinity` are pointers. We need nil-check here to defend NPE panic.

Reply:
yes, but RequiredDuringSchedulingIgnoredDuringExecution, etc, is not, so you don't need to check that for nil